src/plots/layout_attributes.js

```diff
@@ -444,6 +445,112 @@ module.exports = {
         editType: 'modebar'
     },
 
+    newshape: {
```

Comment:
Can we define these attributes somewhere in `components/shapes` and import them here?

Reply:
Good call. Addressed in https://github.com/plotly/plotly.js/pull/4775/commits/63ef7aaf311858101c7ed3075a40472eb6341cd1.